.github/api_best_practices.md

```diff
@@ -0,0 +1,231 @@
+# ACS API Guidelines
+An overview of the set of standardized guidelines and practices to ensure that the APIs across different 
+projects and teams adhere to a common structure and design. This consistency simplifies the process of understanding 
+and maintaining APIs. The guidelines helps identify potential pitfalls and design choices that could lead to issues to 
+help developers avoid common mistakes and build more robust APIs. When in doubt, It is encouraged to seek resolution
+in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archives/C05MMG2PP8A).
+
+## Table of Contents
+
+- [Naming Guidelines](#naming-guidelines)
+  - [gRPC Service Name](#grpc-service-name)
+  - [gRPC Method Name](#grpc-method-name)
+  - [gRPC Message Name](#grpc-message-name)
+
+## Naming Guidelines
+
+### gRPC Service Name
+Service name **must** be unique and should use a noun that generally refers to a resource or product component 
+e.g. `DeploymentService`, `ReportService`, `ComplianceService`. Intuitive and well-known short forms or 
+abbreviations may be used in some cases (and could even be preferable) for succinctness 
+e.g. `ReportConfigService`, `RbacService`. 
```

Comment:
Not sure if this is too specific, but should we define case conventions? e.g. Should it be `RbacService` or `RBACService`?  (Sorry if this is some implied gRPC convention already.)

Reply:
It seems too specific to me. We have to let engineers use their best judgment.